### DISCUSSION TITLE
IAS4SignalMessageConsumer: Add possibility to validate the signal (Signature)

### DISCUSSION THREAD

**Discussion by sopgreg** [phax/phase4#176] [General]
Similiar to https://github.com/phax/phase4/issues/162, it would be helpful to have a possibility to validate the Signal/Receipt received on the outgoing side, e.g. through the `SignalMessageConsumer`

Right now, the signature of `IAS4SignalMessageConsumer#handleSignalMessage` looks like this:

`void handleSignalMessage (@Nonnull Ebms3SignalMessage aSignalMsg) throws Phase4Exception;`

If the signature looked like this, it would be possible to check the message state for `isSoapSignatureChecked`/`isSoapDecrypted`:

`void handleSignalMessage (@Nonnull Ebms3SignalMessage aSignalMsg, @Nonnull IAS4MessageState state) throws Phase4Exception;`

Technically, this should be possible quite easily by changing the result of `com.helger.phase4.servlet.AS4IncomingHandler#parseSignalMessage` to return both the `Ebms3SignalMessage` and the `IAS4MessageState` and return this to the `SignalMessageConsumer`.

Looking forward to reading your opinion in this requirement.

**Comment by phax** [comment]
The `IAS4SignalMessageConsumer` and `IAS4UserMessageConsumer` APIs were extended. That's a backward incompatible change, that's why it is introduced in version 2.5.0.
